[openimageio] Build error - with VS 2017

**Host Environment**

- OS: Windows
- Compiler: Visual Studio 2017 64bit

**To Reproduce**

1) ./vcpkg install openimageio:x64-windows

if later version of Visual Studio installed, 
add: `set(VCPKG_PLATFORM_TOOLSET v141)`
to: `vcpkg\triplets\x64-windows.cmake`

**Failure logs**

```
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:131 (message):
    Command failed: D:/vcpkg/vcpkg/downloads/tools/cmake-3.25.0-windows/cmake-3.25.0-windows-i386/bin/cmake.exe --build . --config Debug --target install -- -v -j13
    Working Directory: D:/vcpkg/vcpkg/buildtrees/openimageio/x64-windows-dbg
    See logs for more information:
      D:\vcpkg\vcpkg\buildtrees\openimageio\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/openimageio/portfile.cmake:71 (vcpkg_cmake_install)
  scripts/ports.cmake:147 (include)
```
From install-x64-windows-dbg-out.log
```
[26/174] cmd.exe /C "cd . && D:\vcpkg\vcpkg\downloads\tools\cmake-3.25.0-windows\cmake-3.25.0-windows-i386\bin\cmake.exe -E vs_link_dll --intdir=src\libutil\CMakeFiles\OpenImageIO_Util.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100177~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100177~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~2\2017\PROFES~1\VC\Tools\MSVC\1416~1.270\bin\Hostx64\x64\link.exe  src\libutil\CMakeFiles\OpenImageIO_Util.dir\argparse.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\benchmark.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\errorhandler.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\farmhash.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\filesystem.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\fmath.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\filter.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\hashes.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\paramlist.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\plugin.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\SHA1.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\strutil.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\sysutil.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\thread.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\timer.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\typedesc.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\ustring.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\xxhash.cpp.obj  /out:bin\OpenImageIO_Util_d.dll /implib:lib\OpenImageIO_Util_d.lib /pdb:bin\OpenImageIO_Util_d.pdb /dll /version:2.4 /machine:x64 /nologo    /debug /INCREMENTAL -LIBPATH:D:\vcpkg\vcpkg\installed\x64-windows\lib   -LIBPATH:D:\vcpkg\vcpkg\installed\x64-windows\debug\lib D:\vcpkg\vcpkg\installed\x64-windows\debug\lib\Imath-3_1_d.lib  D:\vcpkg\vcpkg\installed\x64-windows\debug\lib\boost_filesystem-vc140-mt-gd.lib  D:\vcpkg\vcpkg\installed\x64-windows\debug\lib\boost_thread-vc140-mt-gd.lib  psapi.lib  D:\vcpkg\vcpkg\installed\x64-windows\debug\lib\boost_chrono-vc140-mt-gd.lib  D:\vcpkg\vcpkg\installed\x64-windows\debug\lib\boost_atomic-vc140-mt-gd.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib  && cd ."
FAILED: bin/OpenImageIO_Util_d.dll lib/OpenImageIO_Util_d.lib 
cmd.exe /C "cd . && D:\vcpkg\vcpkg\downloads\tools\cmake-3.25.0-windows\cmake-3.25.0-windows-i386\bin\cmake.exe -E vs_link_dll --intdir=src\libutil\CMakeFiles\OpenImageIO_Util.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100177~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100177~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~2\2017\PROFES~1\VC\Tools\MSVC\1416~1.270\bin\Hostx64\x64\link.exe  src\libutil\CMakeFiles\OpenImageIO_Util.dir\argparse.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\benchmark.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\errorhandler.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\farmhash.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\filesystem.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\fmath.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\filter.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\hashes.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\paramlist.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\plugin.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\SHA1.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\strutil.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\sysutil.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\thread.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\timer.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\typedesc.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\ustring.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\xxhash.cpp.obj  /out:bin\OpenImageIO_Util_d.dll /implib:lib\OpenImageIO_Util_d.lib /pdb:bin\OpenImageIO_Util_d.pdb /dll /version:2.4 /machine:x64 /nologo    /debug /INCREMENTAL -LIBPATH:D:\vcpkg\vcpkg\installed\x64-windows\lib   -LIBPATH:D:\vcpkg\vcpkg\installed\x64-windows\debug\lib D:\vcpkg\vcpkg\installed\x64-windows\debug\lib\Imath-3_1_d.lib  D:\vcpkg\vcpkg\installed\x64-windows\debug\lib\boost_filesystem-vc140-mt-gd.lib  D:\vcpkg\vcpkg\installed\x64-windows\debug\lib\boost_thread-vc140-mt-gd.lib  psapi.lib  D:\vcpkg\vcpkg\installed\x64-windows\debug\lib\boost_chrono-vc140-mt-gd.lib  D:\vcpkg\vcpkg\installed\x64-windows\debug\lib\boost_atomic-vc140-mt-gd.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib  && cd ."
LINK Pass 1: command "C:\PROGRA~2\MICROS~2\2017\PROFES~1\VC\Tools\MSVC\1416~1.270\bin\Hostx64\x64\link.exe src\libutil\CMakeFiles\OpenImageIO_Util.dir\argparse.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\benchmark.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\errorhandler.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\farmhash.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\filesystem.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\fmath.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\filter.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\hashes.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\paramlist.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\plugin.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\SHA1.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\strutil.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\sysutil.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\thread.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\timer.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\typedesc.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\ustring.cpp.obj src\libutil\CMakeFiles\OpenImageIO_Util.dir\xxhash.cpp.obj /out:bin\OpenImageIO_Util_d.dll /implib:lib\OpenImageIO_Util_d.lib /pdb:bin\OpenImageIO_Util_d.pdb /dll /version:2.4 /machine:x64 /nologo /debug /INCREMENTAL -LIBPATH:D:\vcpkg\vcpkg\installed\x64-windows\lib -LIBPATH:D:\vcpkg\vcpkg\installed\x64-windows\debug\lib D:\vcpkg\vcpkg\installed\x64-windows\debug\lib\Imath-3_1_d.lib D:\vcpkg\vcpkg\installed\x64-windows\debug\lib\boost_filesystem-vc140-mt-gd.lib D:\vcpkg\vcpkg\installed\x64-windows\debug\lib\boost_thread-vc140-mt-gd.lib psapi.lib D:\vcpkg\vcpkg\installed\x64-windows\debug\lib\boost_chrono-vc140-mt-gd.lib D:\vcpkg\vcpkg\installed\x64-windows\debug\lib\boost_atomic-vc140-mt-gd.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:src\libutil\CMakeFiles\OpenImageIO_Util.dir/intermediate.manifest src\libutil\CMakeFiles\OpenImageIO_Util.dir/manifest.res" failed (exit code 1120) with the following output:
   Creating library lib\OpenImageIO_Util_d.lib and object lib\OpenImageIO_Util_d.exp
strutil.cpp.obj : error LNK2001: unresolved external symbol "__declspec(dllimport) public: static class std::locale::id std::codecvt<char16_t,char,struct _Mbstatet>::id" (__imp_?id@?$codecvt@_SDU_Mbstatet@@@std@@2V0locale@2@A)
bin\OpenImageIO_Util_d.dll : fatal error LNK1120: 1 unresolved externals
```
**Additional context**

Build succeeded with VS 2022, consistently failed with VS 2017.

[install-x64-windows-dbg-out_1.log](https://github.com/microsoft/vcpkg/files/10349185/install-x64-windows-dbg-out_1.log)

--- a/ports/openimageio/fix-vs2019-encoding-conversion.patch
+++ b/ports/openimageio/fix-vs2019-encoding-conversion.patch
@@ -1,0 +1,20 @@
+diff --git a/src/libutil/strutil.cpp b/src/libutil/strutil.cpp
+index 129a9b5..09baea8 100644
+--- a/src/libutil/strutil.cpp
++++ b/src/libutil/strutil.cpp
+@@ -914,8 +914,15 @@ std::string
+ Strutil::utf16_to_utf8(const std::u16string& str) noexcept
+ {
+     try {
++        // https://stackoverflow.com/a/35103224
++#if defined _MSC_VER && _MSC_VER >= 1900 && _MSC_VER < 1930
++        std::wstring_convert<std::codecvt_utf8_utf16<int16_t>, int16_t> convert;
++        auto p = reinterpret_cast<const int16_t *>(str.data());
++        return convert.to_bytes(p, p + str.size());
++#else
+         std::wstring_convert<std::codecvt_utf8_utf16<char16_t>, char16_t> conv;
+         return conv.to_bytes(str);
++#endif
+     } catch (const std::exception&) {
+         return std::string();
+     }

--- a/ports/openimageio/portfile.cmake
+++ b/ports/openimageio/portfile.cmake
@@ -12,6 +12,7 @@ vcpkg_from_github(
         fix-openexr-dll.patch
         imath-version-guard.patch
         fix-openimageio_include_dir.patch
+        fix-vs2019-encoding-conversion.patch
 )
 
 file(REMOVE_RECURSE "${SOURCE_PATH}/ext")

--- a/ports/openimageio/vcpkg.json
+++ b/ports/openimageio/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "openimageio",
   "version": "2.4.5.0",
+  "port-version": 1,
   "description": "A library for reading and writing images, and a bunch of related classes, utilities, and application.",
   "homepage": "https://github.com/OpenImageIO/oiio",
   "license": "BSD-3-Clause",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5590,7 +5590,7 @@
     },
     "openimageio": {
       "baseline": "2.4.5.0",
-      "port-version": 0
+      "port-version": 1
     },
     "openjpeg": {
       "baseline": "2.5.0",

--- a/versions/o-/openimageio.json
+++ b/versions/o-/openimageio.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "69b98b0baa3eddf9f9552563b39561ed09df2971",
+      "version": "2.4.5.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "74ab484049a78fea2929dc9afc4ae5a6b9b2dd45",
       "version": "2.4.5.0",
       "port-version": 0

--- a/versions/o-/openimageio.json
+++ b/versions/o-/openimageio.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "69b98b0baa3eddf9f9552563b39561ed09df2971",
+      "git-tree": "ecd81b1df58bf0fb628ad86bfde7522e36a602e4",
       "version": "2.4.5.0",
       "port-version": 1
     },

